imports.go

```diff
@@ -30,60 +30,66 @@ type ImportedData struct {
 }
 
 // An Importer imports data from a path.
+// TODO(sbarzowski) caching of errors (may require breaking changes)
 type Importer interface {
-	Import(codeDir string, importedPath string) (*ImportedData, error)
+	// Import fetches data from a given path. It may be relative
+	// to the file where we do the import. What "relative path"
+	// means depends on the importer.
+	//
+	// It is required that:
+	// a) for given (from, path) the same (contents, foundAt) are
+	//    returned on subsequent calls.
+	// b) for given foundAt, the contents are always the same
+	//
+	// Note that by "the same contents" we mean the same pointer and
```

Comment:
It is a little bit weird (if you haven't read the implementation) to have to return the same pointer.  We could alternatively say that "If you return the same foundAt for multiple imports, we use the content for the first one."  That would mean cacheing the LiteralString in the codeCache.

On the other hand, it's not a big deal for implementers to actually meet these requirements, so I'm happy either way.